src/main/java/Deadline.java

```diff
@@ -1,16 +1,19 @@
 import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
 
 public class Deadline extends Task implements Serializable {
 
-    protected String by;
+    protected LocalDate by;
 
-    public Deadline(String description, String by) {
+    public Deadline(String description, LocalDate by) {
         super(description);
         this.by = by;
     }
 
     @Override
     public String toString() {
-        return "[D]" + super.toString() + " (by: " + by + ")";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
```

Comment:
What about the time? How does the bot accept the time of the deadline / event?

Reply:
If im not wrong the requirements said use LocalDate, but LocalDate cannot store time values